src/definitions/elements/input.less

```diff
@@ -455,47 +455,40 @@
   margin: 0;
 }
 
-/* Button on Right */
-.ui.action.input:not([class*="left action"]) > input {
-  border-top-right-radius: 0 !important;
-  border-bottom-right-radius: 0 !important;
-  border-right-color: transparent !important;
-}
-.ui.action.input:not([class*="left action"]) > .dropdown:not(:first-child),
-.ui.action.input:not([class*="left action"]) > .button:not(:first-child),
-.ui.action.input:not([class*="left action"]) > .buttons:not(:first-child) > .button {
-  border-radius: 0;
-}
-.ui.action.input:not([class*="left action"]) > .dropdown:last-child,
-.ui.action.input:not([class*="left action"]) > .button:last-child,
-.ui.action.input:not([class*="left action"]) > .buttons:last-child > .button {
-  border-radius: 0 @borderRadius @borderRadius 0;
+/* Input when ui Left*/
+.ui[class*="left action"].input > input {
+  border-top-left-radius: 0;
+  border-bottom-left-radius: 0;
+  border-left-color: transparent;
 }
 
-/* Input Focus */
-.ui.action.input:not([class*="left action"]) > input:focus {
-  border-right-color: @focusBorderColor !important;
+/* Input when ui Right*/
+.ui.action.input:not([class*="left action"]) > input {
+  border-top-right-radius: 0;
+  border-bottom-right-radius: 0;
+  border-right-color: transparent;
 }
 
-/* Button on Left */
-.ui[class*="left action"].input > input {
-  border-top-left-radius: 0 !important;
-  border-bottom-left-radius: 0 !important;
-  border-left-color: transparent !important;
+/* Button and Dropdown */
+.ui.action.input > .dropdown:first-child,
+.ui.action.input > .button:first-child,
+.ui.action.input > .buttons:first-child > .button {
+  border-radius: @borderRadius 0 0 @borderRadius;
 }
-.ui[class*="left action"].input > .dropdown,
-.ui[class*="left action"].input > .button,
-.ui[class*="left action"].input > .buttons > .button {
+.ui.action.input > .dropdown:not(:first-child),
+.ui.action.input > .button:not(:first-child),
+.ui.action.input > .buttons:not(:first-child) > .button {
   border-radius: 0;
 }
-.ui[class*="left action"].input > .dropdown:first-child,
-.ui[class*="left action"].input > .button:first-child,
-.ui[class*="left action"].input > .buttons:first-child > .button {
-  border-radius: @borderRadius 0 0 @borderRadius;
+.ui.action.input > .dropdown:last-child,
+.ui.action.input > .button:last-child,
+.ui.action.input > .buttons:last-child > .button {
+  border-radius: 0 @borderRadius @borderRadius 0;
 }
+
 /* Input Focus */
-.ui[class*="left action"].input > input:focus {
-  border-left-color: @focusBorderColor !important;
+.ui.ui.action.input:not([class*="left action"]) > input:focus {
+  border-right-color: @focusBorderColor;
 }
```

Comment:
Sorry, but that was the wrong code line (although its good, you removed the !Important there aswell)


There is an additional `/* Input Focus */` existing, which (additional `.ui` only need for the second declaration!
In the end we need to have 
```css
/* Input Focus */

.ui.action.input:not([class*="left action"]) > input:focus {
  border-right-color: #85B7D9;
}
.ui.ui[class*="left action"].input > input:focus {
  border-left-color: #85B7D9;
}
```

Reply:
Committed that.